packages/mobile/src/utils/time.ts

```diff
@@ -318,3 +320,29 @@ function quickFormat(timestamp: number, i18next: i18nType, formatRule: string) {
     locale: i18next.language.includes('es') ? es : enUS,
   })
 }
+
+// update moving average for time ellapsed
+export function updateMeanMillisecs(meanMillisecs: number, millisecs: number, onceEvery: number) {
+  const multiplier = meanMillisecs > 0.0 ? meanMillisecs / onceEvery : 1.0
+  meanMillisecs = meanMillisecs * (1.0 - multiplier) + multiplier * millisecs
+  return meanMillisecs
+}
+
+// warning: this function has side-effects on module
+//          vars meanMillisecs and trackCount.
+export const trackMeanMillisecs = (
+  initTime: number,
+  meanMillisecs: number,
+  trackCount: number,
+  onceEvery: number,
+  eventName: CustomEventNames,
+  attachDeviceInfo: boolean = false
```

Comment:
this param isn't used anywhere is it?

Reply:
Yes, on CustomEventNames.sync_blockchain_corruption and CustomEventNames.sync_unexpected_error.